app/preload/ledger/ledgerObserverImpl.ts

```diff
@@ -10,22 +10,75 @@ import type {
 import EventEmitter from 'events';
 import { usb } from 'usb';
 import { identifyUSBProductId, ledgerUSBVendorId } from '@ledgerhq/devices';
+import { Mutex } from 'async-mutex';
+import fs from 'fs';
+import path from 'path';
 import ConcordiumLedgerClientMain from '../../features/ledger/ConcordiumLedgerClientMain';
 import { isConcordiumApp, isOutdated } from '../../components/ledger/util';
 import { LedgerSubscriptionAction } from '../../components/ledger/useLedger';
 import ledgerIpcCommands from '~/constants/ledgerIpcCommands.json';
 import { LedgerObserver } from './ledgerObserver';
 
+function logError(message: string, ...args: unknown[]) {
+    const logPath = path.join(__dirname, 'crash.log');
+    const timestamp = new Date().toISOString();
+    const details = args.map(a => {
+        if (a instanceof Error) {
+            return `${a.name}: ${a.message}\n${a.stack}`;
+        }
+        try {
+            return JSON.stringify(a);
+        } catch {
+            return String(a);
+        }
+    }).join(' ');
+    const fullMsg = `[${timestamp}] ERROR: ${message} ${details}\n`;
+    try {
+        fs.appendFileSync(logPath, fullMsg);
+    // eslint-disable-next-line no-empty
+    } catch (e) {}
+
+}
```

Comment:
Seems very ad-hoc, we already have `winston` installed for logging.. can we not use this? it's available through `preload/logging.ts`